### PR TITLE
[FIX] hr_calendar : fix unavailable partner computation

### PR DESCRIPTION
**Pull request by mepe-odoo**
Issue 1:

    Step to reproduce:
    ==================
    
        1- Go on calendar application
        2- Go on List view
        3- Click on the event "Changes in Designing"
    
    A traceback will be raised

    Expected Behaviour:
    ===================
        This event is an allday event during a worked day for the company.
        So a partner will be available only if he works during the company's
        interval for this day.
    
    Issue explanation:
    ==================
        The event "Changes in Designing" is an all day event with a
        start_date equals to the stop_date. From the front_end with the
        method _inverse_dates, the sart_date will be equals to 8AM and the
        stop_date will be equals to 6PM.
        But this event is created with demo data; so the inverse method is
        not called.
        And, to be able to synchronise calendar application with google
        calendar and outlook calendar; it's mandatory to allow the user to
        create an event with a duration of 0.

Issue 2:

    STEP TO REPRODUCE:
    =================
        1- Go on calendar Application
        2- Click on sunday all day box
    
    You will have a traceback about unavailable partner computation.
    
    Expect Behaviour:
    =================
    If the employee is not supposed to work during an entire day; when you
    want to create an all day event during this day the employee's partner
    will be unavailable.

task-4042363

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_calendar/tests/test_event_interval.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            )
+        ])
+
+    def test_allday_event_during_no_working_day(self):
+        event = self.env['calendar.event'].with_context(company_id=self.company_A.id).create([
+            {
+                'start': datetime(2024, 7, 13),
+                'stop': datetime(2024, 7, 13, 23, 59, 59),
+                'allday': True,
+                'name': "Event 2"
```
Comment (Flotchet) [inline addons/hr_calendar/tests/test_event_interval.py:45]: '

File: addons/hr_calendar/tests/test_event_interval.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        super().setUpClass()
+        cls.company_A.resource_calendar_id = cls.calendar_35h
+
+    def test_allday_event_during_working_day(self):
+        event = self.env['calendar.event'].with_context(company_id=self.company_A.id).create([
+            {
+                'start': datetime(2024, 7, 12),
+                'stop': datetime(2024, 7, 12, 23, 59, 59),
+                'allday': True,
+                'name': "Event 1"
```
Comment (Flotchet) [inline addons/hr_calendar/tests/test_event_interval.py:22]: ' ? R----------------------------------------------

especially if you want to do 

```py
                'tz': "Europe/Brussels",
                'tz': 'Europe/Brussels',
```

File: addons/hr_calendar/tests/test_event_interval.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                self.env['resource.calendar']
+            )
+        ])
+
+    def test_event_during_working_day(self):
+        event = self.env['calendar.event'].with_context(company_id=self.company_A.id).create([
+            {
+                'start': datetime(2024, 7, 13, 8, 30, 0),
+                'stop': datetime(2024, 7, 13, 9, 30, 0),
+                'name': "Event 3"
```
Comment (Flotchet) [inline addons/hr_calendar/tests/test_event_interval.py:62]: '

**Review by Flotchet — APPROVED** [review]
nitpicking so not blocking or important bisous

File: addons/hr_calendar/models/calendar_event.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ def _compute_unavailable_partner_ids(self):
         # based on company calendar's interval.
         for event, event_interval in event_intervals.items():
             start = event_interval._items[0][0]
-            stop = event_interval._items[0][1]
+            stop = event_interval._items[-1][1]
```
Comment (aj-fuentes) [inline addons/hr_calendar/models/calendar_event.py:27]: This doesn't work if `_items` is empty -> current error in runbot.

Comment (mepe-odoo) [reply addons/hr_calendar/models/calendar_event.py:30]: Yep I will add a check :D

Comment (aj-fuentes) [reply addons/hr_calendar/models/calendar_event.py:30]: Well, with all the details we discussed in DM I guess the patch we discussed wouldn't need the check ... but it doesn't hurt :)

Comment (aj-fuentes) [reply addons/hr_calendar/models/calendar_event.py:30]: > I don't know If you see it but I made a commit for your issue ;)

No, I didn't. At first sight it looks good.

**Comment by sofiagvaladze** [comment]
robodoo r+

**Comment by aj-fuentes** [comment]
Hi @sofiagvaladze, could we merge this today?

**Comment by sofiagvaladze** [comment]
@robodoo rebase-ff

**Comment by sofiagvaladze** [comment]
robodoo rebase-ff

**Pull request closed without merging**